Cython/Compiler/Parsing.py

```diff
@@ -4593,25 +4593,21 @@ def p_group_pattern(s: PyrexScanner):
 
 @cython.cfunc
 def p_sequence_pattern(s: PyrexScanner):
-    opener = s.sy
     pos = s.position()
-    if opener in ['[', '(']:
-        closer = ']' if opener == '[' else ')'
-        s.next()
-        # maybe_sequence_pattern and open_sequence_pattern
-        patterns = []
-        while s.sy != closer:
-            patterns.append(p_maybe_star_pattern(s))
-            if s.sy == ",":
-                s.next()
-            else:
-                if opener == '(' and len(patterns) == 1:
-                    s.error("tuple-like pattern of length 1 must finish with ','")
-                break
-        s.expect(closer)
-        return MatchCaseNodes.MatchSequencePatternNode(pos, patterns=patterns)
-    else:
-        s.error("Expected '[' or '('")
+    assert s.sy in ('(', '[')
```

Comment:
While you're changing this, there's a comment elsewhere in the file that says

```
* starts with "{" is a sequence_pattern
* starts with "[" is a mapping_pattern
```

which is the wrong way around (and almost certainly my fault). But might be worth correcting.